client-sdk/ts-web/ext-utils/src/protocol.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file Constants and type definitions for the `postMessage`-based protocol
+ * used between web content and the extension.
+ */
+
+// page <- message frame
+export const MESSAGE_TYPE_READY = 'oasis-xu-ready';
+export const MESSAGE_TYPE_RESPONSE = 'oasis-xu-response';
+
+export interface MessageReady {
+    type: typeof MESSAGE_TYPE_READY;
+}
+
+export interface MessageResponse {
+    type: typeof MESSAGE_TYPE_RESPONSE;
+    id: number;
+    body?: unknown;
+    err?: unknown;
+}
+
+// page -> message frame
+export const MESSAGE_TYPE_REQUEST = 'oasis-xu-request';
+
+export interface MessageRequest {
+    type: typeof MESSAGE_TYPE_REQUEST;
+    id: number;
+    body: unknown;
+}
+
+// methods
+
+export const METHOD_KEYS_LIST = 'list-keys';
```

Comment:
```suggestion
export const METHOD_KEYS_LIST = 'keys-list';
```
Either that or rename all the types/vars.